src/coreclr/System.Private.CoreLib/src/System/Array.CoreCLR.cs

```diff
@@ -381,8 +381,57 @@ private unsafe bool IsValueOfElementType(object value)
         // if this is an array of value classes and that value class has a default constructor
         // then this calls this default constructor on every element in the value class array.
         // otherwise this is a no-op.  Generally this method is called automatically by the compiler
-        [MethodImpl(MethodImplOptions.InternalCall)]
-        public extern void Initialize();
+        public unsafe void Initialize()
```

Comment:
```suggestion
        [RequiresUnreferenced("The public parameterless constructor may be trimmed from the array element type.")]
        public unsafe void Initialize()
```
(same for all implementations of this method)